lib/browser/ipc-main-internal-utils.ts

```diff
@@ -1,26 +1,16 @@
 import { ipcMainInternal } from '@electron/internal/browser/ipc-main-internal'
 import * as errorUtils from '@electron/internal/common/error-utils'
 
-type IPCHandler = (event: ElectronInternal.IpcMainInternalEvent, ...args: any[]) => any
-
-const callHandler = async function (handler: IPCHandler, event: ElectronInternal.IpcMainInternalEvent, args: any[], reply: (args: any[]) => void) {
-  try {
-    const result = await handler(event, ...args)
-    reply([null, result])
-  } catch (error) {
-    reply([errorUtils.serialize(error)])
-  }
-}
+type IPCHandler = (event: Electron.IpcMainInvokeEvent, ...args: any[]) => any
 
 export const handle = function <T extends IPCHandler> (channel: string, handler: T) {
-  ipcMainInternal.on(channel, (event, requestId, ...args) => {
-    callHandler(handler, event, args, responseArgs => {
-      if (requestId) {
-        event._replyInternal(`${channel}_RESPONSE_${requestId}`, ...responseArgs)
-      } else {
-        event.returnValue = responseArgs
-      }
-    })
+  ipcMainInternal.handle(channel, handler)
+  ipcMainInternal.on(channel, async (event, ...args) => {
```

Comment:
Registering two separate handlers seems like a bad idea to me. Can we just call `ipcMainInternal.handle` directly for when we want to call with `invoke`?

Reply:
done